File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func rightSideView(root *TreeNode) []int {
	if root == nil {
		return nil
	}
    ans := make([]int, 0, 100)
	q := make([]*TreeNode,0,100)
	q = append(q, root)
	for len(q) > 0 {
		sz := len(q)
		for i:=0; i<sz; i++ {
			cur := q[0]
			q[0] = nil
			q = q[1:]
			if cur.Left != nil {
				q = append(q, cur.Left)
			}
			if cur.Right != nil {
				q = append(q, cur.Right)
			}
			if i==sz-1{
				ans = append(ans,cur.Val)
			}
		}
	}
	return ans
}

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int kthSmallest(TreeNode root, int k) {
		Deque<TreeNode> stack = new ArrayDeque<>();
		var cur = root;
		while( cur != null || !stack.isEmpty()){
			while(cur != null){
				stack.push(cur);
				cur = cur.left;
			}
			var n = stack.pop();
			if(--k==0) return n.val;
			cur = n.right;
		}
		return -1;
    }
}
